internal/querynodev2/delegator/delegator.go

```diff
@@ -662,10 +668,71 @@ func (sd *shardDelegator) Close() {
 	sd.lifetime.Wait()
 }
 
+// As partition stats is an optimization for search/query which is not mandatory for milvus instance,
+// loading partitionStats will be a try-best process and will skip+logError when running across errors rather than
+// return an error status
+func (sd *shardDelegator) maybeReloadPartitionStats(ctx context.Context, partIDs ...UniqueID) {
+	var partsToReload []UniqueID
+	if len(partIDs) > 0 {
+		partsToReload = partIDs
+	} else {
+		partsToReload = append(partsToReload, sd.collection.GetPartitions()...)
+	}
+
```

Comment:
we will list all available part_stats files and find the file with max version number